src/Tribe/Views/V2/View.php

```diff
@@ -2194,50 +2194,77 @@ protected function should_show_recent_past_events_view() {
 	 */
 	protected function setup_additional_views( array $events = [], array $template_vars = [] ) {
 
-		$manager = tribe( Manager::class );
+		$manager      = tribe( Manager::class );
 		$default_slug = $manager->get_default_view_option();
 
-		// Only recent Past Events to only the default view.
+		// Show Latest Past Events only on the default view.
 		if ( $this->get_slug() !== $default_slug ) {
 			return;
 		}
 
+		$now    = $this->context->get( 'now' );
+		$latest = tribe_events_latest_date();
+
+		// If now is less then the latest event published, do not show.
+		if ( $now < $latest ) {
+			return;
+		}
+
+
+		// Checks to verify on the initial load of a view or if using today's date for the view.
+		$today     = $this->context->get( 'today' );
+		$view_date = $this->context->get( 'event_date', '' );
+		if ( 'month' === $this->get_slug() ) {
+			$today_formatted     = Dates::build_date_object( $today )->format( Dates::DBYEARMONTHTIMEFORMAT );
+			$view_date_formatted = Dates::build_date_object( $view_date )->format( Dates::DBYEARMONTHTIMEFORMAT );
+		} elseif ( 'week' === $this->get_slug() ) {
```

Comment:
should this be filtered so that ECP can hook into and do its stuff there?

Reply:
If we filter here, then we should filter for all Views, not just week
In other parts of this code there are general and view-specific filters to control the view; I think that is enough for now.
If we find we need a filter here, though, we can always add it in the future.